rct229/rulesets/ashrae9012019/ruleset_functions/get_zone_target_baseline_system.py

```diff
@@ -0,0 +1,244 @@
+    expected_system_type_from_table_g3_1_1_dict,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.g311_exceptions.g311_sub_functions.get_computer_zones_peak_cooling_load import (
+    get_computer_zones_peak_cooling_load,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.g311_exceptions.g311_sub_functions.get_hvac_building_area_types_and_zones_dict import (
+    get_hvac_building_area_types_and_zones_dict,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.g311_exceptions.g311_sub_functions.get_number_of_floors import (
+    get_number_of_floors,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.g311_exceptions.g311_sub_functions.get_predominant_hvac_building_area_type import (
+    get_predominant_hvac_building_area_type,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.g311_exceptions.g311_sub_functions.get_zone_hvac_bat import (
+    get_zone_hvac_bat_dict,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_zone_conditioning_category_dict import (
+    ZoneConditioningCategory as ZCC,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_zone_conditioning_category_dict import (
+    get_zone_conditioning_category_dict,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.is_cz_0_to_3a_bool import (
+    is_cz_0_to_3a_bool,
+)
+from rct229.schema.config import ureg
+from rct229.utils.assertions import assert_
+from rct229.utils.jsonpath_utils import find_all
+
+BUILDING_AREA_20000_ft2 = 20000 * ureg("ft2")
+BUILDING_AREA_40000_ft2 = 40000 * ureg("ft2")
+BUILDING_AREA_150000_ft2 = 150000 * ureg("ft2")
+REQ_FL_6 = 6
+COMPUTER_ROOM_PEAK_COOLING_LOAD_600000_BTUH = 600000 * ureg("Btu/hr")
+COMPUTER_ROOM_PEAK_COOLING_LOAD_3000000_BTUH = 3000000 * ureg("Btu/hr")
+
+
+class SYSTEMORIGIN:
+    G311B = "G3_1_1b"
+    G311C = "G3_1_1c"
+    G311D = "G3_1_1d"
+    G311E = "G3_1_1e"
+    G311F = "G3_1_1f"
+
+
+def get_zone_target_baseline_system(rmd_b: dict, rmd_p: dict, climate_zone_b: str):
+    """
+    Following G3.1.1, determines the baseline system type for each zone in a building
+
+    Parameters
+    ----------
+    rmd_b: json
+        RMD at RuleSetModelDescription level
+    rmd_p: json
+        RMD at RuleSetModelDescription level
+
+    climate_zone_b: str
+        baseline climate zone
+
+    Returns
+    -------
+    zones_and_systems: a dictionary with zone / list pairs where the first value in the list is the expected system type (ex SYS_3) and the second value is the rule used to choose the system, (eg "G3_1_1e"): zones_and_systems[zone]["EXPECTED_SYSTEM_TYPE"] = SYS_3; zones_and_systems[zone]["SYSTEM_ORIGIN"] = "G3_1_1e"
+
+    """
+
+    # make sure `buildings` key exists
+    assert_(rmd_b["buildings"][0], "Please make sure the `buildings` key exists.")
+
+    building_b = rmd_b["buildings"][0]
+    zone_conditioning_category_dict = get_zone_conditioning_category_dict(
```

Comment:
This function is RMI level, we do not need to take out a specific building from the RMI.
You can use `get_zone_conditioning_category_rmi_dict(climate_zone, rmi)` instead of `get_zone_conditioning_category_dict`

Reply:
Great point! Addressed.